### PR TITLE
feat: universal resource internal sharing

### PR DESCRIPTION
**Pull request by 3Hren**
Goals:
- [x] Provide available resources.
- [x] Optional configurable restrictions.
- [x] Push notification.
- [x] Nodes discovery (alongside with locator) for scheduling.
- [x] Debianization.

### CODE REVIEW

**Comment by 3Hren** [comment]
@cocaine-plugins TAL.

**Comment by 3Hren** [comment]
@antmat @karitra PTAL.

File: CMakeLists.txt [hunk 1]
```diff
@@ -72,6 +73,7 @@ SET(PLUGINS
     postgres
     unicat
     unicorn
+	uniresis
```
Comment (karitra) [inline CMakeLists.txt]: style? extra tab/spaces?

File: uniresis/include/cocaine/service/uniresis.hpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+#include <cocaine/api/service.hpp>
+#include <cocaine/rpc/dispatch.hpp>
+
+#include "cocaine/idl/uniresis.hpp"
+#include "cocaine/uniresis/resources.hpp"
+
+namespace cocaine {
+namespace service {
+
+class uniresis_t : public api::service_t , dispatch<io::uniresis_tag> {
```
Comment (karitra) [inline uniresis/include/cocaine/service/uniresis.hpp]: Is it ok for dispatch<> to be without access specifier?

File: uniresis/src/uniresis.cpp [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+        );
+    }
+
+private:
+    auto
+    make_value() -> dynamic_t {
+        dynamic_t::object_t result;
+
+        std::vector<dynamic_t> endpoints;
+        for (auto endpoint : this->endpoints) {
```
Comment (karitra) [inline uniresis/src/uniresis.cpp]: Possible would be optimized, but may be use 'const auto& ref'?

**Review by karitra — APPROVED** [review]
LGTM

**Comment by antmat** [comment]
We need to check if ephemeral node has gone

**Comment by 3Hren** [comment]
Rereview please.

**Review by antmat — APPROVED** [review]
nice

**Pull request merged**